### ISSUE TITLE
ContractConfigurator issue still persists

### ISSUE THREAD

**Issue by Gordon-Dry** [steamp0rt/Kerbalism#9]
I just copy this issue
https://github.com/ScienceFun/Kerbalism-Continued/issues/9
to not be forgotten, because it still exists.

The ModuleManager.ConfigCache will never be created because:
```
[ERR 17:31:32.958] [ModuleManager] Exception while processing node : Kerbalism/Support/ContractConfigurator/@CONTRACT_TYPE[*]

[EXC 17:31:32.961] NullReferenceException: Object reference not set to an instance of an object
	ConfigNode.CopyToRecursive (.ConfigNode node, Boolean overwrite)
	ConfigNode.CopyToRecursive (.ConfigNode node, Boolean overwrite)
	ConfigNode.CopyTo (.ConfigNode node, System.String newName)
	ModuleManager.MMPatchLoader.ModifyNode (ModuleManager.Collections.ImmutableStack`1 original, .ConfigNode mod, PatchContext context)
	ModuleManager.PatchApplier.ApplyPatches (System.String stage, IEnumerable`1 patches)
	UnityEngine.Logger:LogException(Exception)
	ModuleManager.Logging.UnityLogger:Exception(String, Exception)
	ModuleManager.Logging.ModLogger:Exception(String, Exception)
	ModuleManager.Logging.ExceptionMessage:LogTo(IBasicLogger)
	ModuleManager.<ProcessPatch>d__33:MoveNext()
	UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[ERR 17:31:32.961] [ModuleManager] Processed node was
Kerbalism/Support/ContractConfigurator/@CONTRACT_TYPE[*]
  @CONTRACT_TYPE[*]
  {
    @PARAMETER[*]:HAS[#partModule[ModuleScienceLab]]
    {
      @partModule = Laboratory
    }
    @PARAMETER[*]
    {
      @PARAMETER[*]:HAS[#partModule[ModuleScienceLab]]
      {
        @partModule = Laboratory
      }
      @PARAMETER[*]
      {
        @PARAMETER[*]:HAS[#partModule[ModuleScienceLab]]
        {
          @partModule = Laboratory
        }
        @PARAMETER[*]
        {
          @PARAMETER[*]:HAS[#partModule[ModuleScienceLab]]
          {
            @partModule = Laboratory
          }
        }
      }
    }
    @REQUIREMENT[*]:HAS[#partModule[ModuleScienceLab]]
    {
      @partModule = Laboratory
    }
    @REQUIREMENT[*]
    {
      @REQUIREMENT[*]:HAS[#partModule[ModuleScienceLab]]
      {
        @partModule = Laboratory
      }
      @REQUIREMENT[*]
      {
        @REQUIREMENT[*]:HAS[#partModule[ModuleScienceLab]]
        {
          @partModule = Laboratory
        }
        @REQUIREMENT[*]
        {
          @REQUIREMENT[*]:HAS[#partModule[ModuleScienceLab]]
          {
            @partModule = Laboratory
          }
        }
      }
    }
  }
```
which results in
```
[LOG 17:31:39.283] [ModuleManager] Done patching
[WRN 17:31:39.296] [ModuleManager] Errors in patch prevents the creation of the cache
[LOG 17:31:39.303] [ModuleManager] ModuleManager: 52663 patches applied, encountered <color=red>1 exception</color>
1 error related to GameData/Kerbalism/Support/ContractConfigurator.cfg

[LOG 17:31:39.304] [ModuleManager] Running tests...
[LOG 17:31:39.306] [ModuleManager] tests complete.
[LOG 17:31:39.307] [ModuleManager] Reloading resources definitions
```

**Comment by madman2003** [comment]
@Gordon-Dry FYI, as a Kerbalism user I have no clue what this is about, so more specific reproduction scenario is probably a good idea ;-)

**Comment by Gordon-Dry** [comment]
@madman2003 Just add Kerbalism when using ContractConfigurator and the error occurs as shown above.

**Comment by madman2003** [comment]
@Gordon-Dry have you tried fixing this yourself or asking the people who made this contract config thing? (this thing seems a bit niche, and doesn't look like any module manager config file i've seen)

**Comment by Gordon-Dry** [comment]
@madman2003 This config belongs to Kerbalism and I already mentioned it in the forum thread some weeks ago, directly pointing at N70.

**Comment by steamp0rt** [comment]
i don't understand what it does

**Comment by Gordon-Dry** [comment]
Me neither.

**Comment by PiezPiedPy** [comment]
I cant reproduce this bug while using ContractConfigurator with Kerbalism but I have made some changes that may fix the issue.

**Comment by PiezPiedPy** [comment]
@Gordon-Dry I suggest you delete ContractConfigurator and Kerbalism GameData directories and install the latest versions if you have not already done so. Then copy https://github.com/PiezPiedPy/Kerbalism/blob/dda112fb22bbf1630926deaf0883f2033577c90f/GameData/Kerbalism/Support/ContractConfigurator.cfg into the Kerbalism/Support folder and let me know if the bug is still present.

**Comment by Gordon-Dry** [comment]
I will do. Just created a copy of my actual save - which is a new test save from yesterday - and removed TAC-LS.

**Comment by Gordon-Dry** [comment]
So, after launching KSP I went to the basement, but coming back I see the game is loaded AND the ModuleManager.ConfigCache is created - so the specific MM issue is solved.
Now I have to proceed to check out new Kerbalism.

**Comment by PiezPiedPy** [comment]
@Gordon-Dry Thanks for letting me know

@steamp0rt This issue can be closed, bug is fixed with PR #25
